### DISCUSSION TITLE
Does i18n global config conflict with on-demand component importing?

### DISCUSSION THREAD

**Discussion by wenfangdu** [element-plus/element-plus#11737] [Q&A] [answered]
It looks like the i18n global config uses full import, if I'm using `unplugin-vue-components`, does the i18n global config negate the benefits of on-demand importing? 

<img width="871" alt="image" src="https://user-images.githubusercontent.com/28700378/221356625-913c9e9e-3d6b-41e0-affc-c9deb0e76c4e.png">
<img width="884" alt="image" src="https://user-images.githubusercontent.com/28700378/221356601-2141216b-9ec7-43b7-9bb7-9a838d8240e2.png">

**Comment by chenxch** [comment, marked as answer]
You can use [config-provider](https://element-plus.org/en-US/component/config-provider.html) with `unplugin-vue-components`.

**Reply by wenfangdu** [reply]
So [this approach](https://element-plus.org/en-US/guide/i18n.html#global-configuration) indeed does full import?

**Reply by chenxch** [reply]
i18n is globally effective.

**Reply by wenfangdu** [reply]
Sorry for not speaking clearly, I meant [this approach](https://element-plus.org/en-US/guide/i18n.html#global-configuration) does import all components from `element-plus`?

**Reply by chenxch** [reply]
This is just i18n, and it doesn't matter whether the components are imported in full.

**Reply by wenfangdu** [reply]
Here's my reasoning, from [here](https://element-plus.org/en-US/guide/quickstart.html#full-import), `app.use(ElementPlus)` imports all components, and from [here](https://element-plus.org/en-US/guide/i18n.html#global-configuration), `app.use(ElementPlus, { locale: zhCn })` registers global language, and they share the same syntax, one would assume that using `app.use(ElementPlus, { locale: zhCn })` would import all components from `element-plus`, am I right? 🥲

**Reply by chenxch** [reply]
right.

**Reply by wenfangdu** [reply]
Thanks.
